console/console-init/ui/src/modules/device/components/AddDeviceWithJson/AddDeviceWithJson.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  PageSection,
+  PageSectionVariants,
+  Flex,
+  FlexItem,
+  FlexModifiers,
+  Button
+} from "@patternfly/react-core";
+import { StyleSheet, css } from "@patternfly/react-styles";
+import { JsonEditor, useWindowDimensions } from "components";
+import {
+  AddJsonUsingTemplate,
+  connectedViaGatewayDeviceTemplate,
+  directlyConnectedDeviceTemplate
+} from "./AddJsonUsingTemplate";
+import { compareJsonObject, getFormattedJsonString } from "utils";
+import { types, MODAL_TYPES, useStoreContext } from "context-state-reducer";
+import { DeviceListAlert } from "modules/device";
+import { TemplateType } from "constant";
+
+const styles = StyleSheet.create({
+  box_align_style: {
+    minHeight: "40em",
+    maxHeight: "40em",
+    border: "1px solid",
+    borderColor: "grey"
+  }
+});
+interface IAddDeviceWithJsonProps {}
+
+const AddDeviceWithJson: React.FunctionComponent<IAddDeviceWithJsonProps> = () => {
+  const { dispatch } = useStoreContext();
+  const width = useWindowDimensions().width;
+  const [json, setJson] = useState<string>();
+  const [selectedTemplate, setSelectedTemplate] = useState<string>(
+    TemplateType.DIRECTLY_CONNECTED
+  );
+  const [showJsonValidationError, setShowJsonValidationError] = useState<
+    boolean
+  >(false);
+
+  const setJsonHandler = (value?: string) => {
+    if (showJsonValidationError) {
+      setShowJsonValidationError(false);
+    }
+    setJson(value);
+  };
+
+  const onLeaveModal = () => {
+    //TODO: handle onConfirm event for modal
+  };
+
+  const onCancel = () => {
+    let isJsonPresent: boolean = false;
+    try {
+      const data = json && JSON.parse(json);
```

Comment:
Not sure about the logic but there are so many ` if else ` conditions .can you revisit this function and try to reduce if else conditions if possible.